addons/mail/static/src/components/composer/composer.js

```diff
@@ -48,35 +48,13 @@ class Composer extends Component {
          * Reference of the text input component.
          */
         this._textInputRef = useRef('textInput');
-        /**
-         * Tracked focus counter from props. Useful to determine whether it
-         * should auto focus this composer when patched.
-         */
-        this._focusCount = 0;
-        // to focus if the prop changes
-        this._lastComposer = undefined;
         this._onClickCaptureGlobal = this._onClickCaptureGlobal.bind(this);
     }
 
     mounted() {
-        this._lastComposer = this.composer;
-        if (this.props.isFocusOnMount) {
-            this.focus();
```

Comment:
This method should be removed, focus should only be called on model never on component.

I don't know what is the case for mobile but it needs to be handled similar to isdofocus

Reply:
component method focus is "justement" doing the model.focus call and handle mobile

```js
    focus() {
        if (this.env.messaging.device.isMobile) {
            this.el.scrollIntoView();
        }
        this.composer.focus();
    }
```

Reply:
The caller should call focus on model directly.

The focus on model should set isDoFocus true OR set another hint for mobile, and component must react to both hints differently.

Reply:
Maybe we should look into that later because I won't have enough time to do it before the merge I think (there is probably the same case with focusout)